presto-main/src/main/java/com/facebook/presto/execution/scheduler/FixedSourcePartitionedScheduler.java

```diff
@@ -67,14 +78,22 @@ public ScheduleResult schedule()
             scheduledTasks = true;
         }
 
-        ScheduleResult schedule = sourcePartitionedScheduler.schedule();
-        return new ScheduleResult(schedule.isFinished(), newTasks, schedule.getBlocked());
+        CompletableFuture<?> blocked = CompletableFuture.completedFuture(null);
+        while (!sourcePartitionedSchedulers.isEmpty()) {
+            ScheduleResult schedule = sourcePartitionedSchedulers.get(0).schedule();
+            blocked = schedule.getBlocked();
+            if (!blocked.isDone() || !schedule.isFinished()) {
```

Comment:
I see. Things start to make sense for me now. This is indeed the case, and that's why we need `SchedulingOrderVisitor`. After all, it is useful to schedule at the same time multiple tablescans (except in limited cases which is not of concern right now)